Description: Install auxiliary binaries in subdirectory of libdir
 Instead of using the root libexec dir folder (/usr/lib/<triplet>), install
 them in its <packagename> subdir.
Author: Nicolas Bourdaud <nicolas.bourdaud@gmail.com>
Origin: vendor
Forwarded: https://github.com/linuxmint/Cinnamon/pull/987
Applied-Upstream: no
Last-Update: 2012-07-29

--- a/src/Makefile-hotplug-sniffer.am
+++ b/src/Makefile-hotplug-sniffer.am
@@ -1,6 +1,6 @@
 service_in_files += hotplug-sniffer/org.Cinnamon.HotplugSniffer.service.in
 
-libexec_PROGRAMS += cinnamon-hotplug-sniffer
+pkglibexec_PROGRAMS += cinnamon-hotplug-sniffer
 
 cinnamon_hotplug_sniffer_SOURCES =								\
 	hotplug-sniffer/hotplug-mimetypes.h \
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -3,7 +3,7 @@
 CLEANFILES =
 EXTRA_DIST =
 bin_SCRIPTS =
-libexec_PROGRAMS =
+pkglibexec_PROGRAMS =
 noinst_LTLIBRARIES =
 noinst_PROGRAMS =
 service_in_files =
@@ -22,7 +22,9 @@
 %.service: %.service.in Makefile
 	$(AM_V_GEN)									\
 		[ -d $(@D) ] || $(mkdir_p) $(@D) ;					\
-		sed -e "s|\@libexecdir\@|$(libexecdir)|" $< > $@.tmp && mv $@.tmp $@
+		sed -e "s|\@libexecdir\@|$(libexecdir)|" \
+		    -e "s|\@pkglibexecdir\@|$(pkglibexecdir)|" $< > $@.tmp && mv $@.tmp $@
+
 CLEANFILES += $(service_DATA)
 
 CLEANFILES += $(gir_DATA) $(typelib_DATA)
@@ -54,6 +56,7 @@
 	    -e "s|@libdir[@]|$(libdir)|" \
 	    -e "s|@JHBUILD_TYPELIBDIR[@]|$(JHBUILD_TYPELIBDIR)|" \
 	    -e "s|@pkgdatadir[@]|$(pkgdatadir)|" \
+	    -e "s|@pkglibexecdir[@]|$(pkglibexecdir)|" \
 	    -e "s|@PYTHON[@]|$(PYTHON)|" \
 	    -e "s|@VERSION[@]|$(VERSION)|" \
 	    -e "s|@sysconfdir[@]|$(sysconfdir)|"
@@ -82,6 +85,7 @@
 	-DCINNAMON_LIBEXECDIR=\"$(libexecdir)\"	\
 	-DCINNAMON_DATADIR=\"$(pkgdatadir)\"	\
 	-DCINNAMON_PKGLIBDIR=\"$(pkglibdir)\" \
+	-DCINNAMON_PKGLIBEXECDIR=\"$(pkglibexecdir)\"	\
 	-DJSDIR=\"$(pkgdatadir)/js\"
 
 privlibdir = $(pkglibdir)
@@ -191,7 +195,7 @@
 
 ########################################
 
-libexec_PROGRAMS += cinnamon-perf-helper
+pkglibexec_PROGRAMS += cinnamon-perf-helper
 
 cinnamon_perf_helper_SOURCES = cinnamon-perf-helper.c
 cinnamon_perf_helper_CPPFLAGS = $(CINNAMON_PERF_HELPER_CFLAGS)
--- a/src/cinnamon-jhbuild.in
+++ b/src/cinnamon-jhbuild.in
@@ -182,7 +182,7 @@
     if running_from_source_tree:
         perf_helper_path = os.path.join(self_dir, "cinnamon-perf-helper")
     else:
-        perf_helper_path = "@libexecdir@/cinnamon-perf-helper"
+        perf_helper_path = "@pkglibexecdir@/cinnamon-perf-helper"
 
     subprocess.Popen([perf_helper_path])
     wait_for_dbus_name (PERF_HELPER_NAME)
--- a/src/hotplug-sniffer/org.Cinnamon.HotplugSniffer.service.in
+++ b/src/hotplug-sniffer/org.Cinnamon.HotplugSniffer.service.in
@@ -1,3 +1,3 @@
 [D-BUS Service]
 Name=org.Cinnamon.HotplugSniffer
-Exec=@libexecdir@/cinnamon-hotplug-sniffer
+Exec=@pkglibexecdir@/cinnamon-hotplug-sniffer
